3_visualize/src/interactive_map.R

```diff
@@ -0,0 +1,56 @@
+# Build interactive map with leaflet
+build_map_leaflet <- function(p3_huc8_sf, p3_saline_lakes_sf, p3_flowlines_sf, p3_gage_sites_sf){
+  # Define color ramp for steams
+  pal_stream <- colorFactor(
+    palette = scico::scico(9, palette = 'davos')[6:3],
+    domain = p3_flowlines_sf$streamorde_size)
+  
+  # Define color ramp for gages
+  pal_gage <- colorFactor(
+    palette = scico::scico(16, palette = 'davos')[c(11, 1)],
+    domain = p3_gage_sites_sf$in_HUC8)
+  
+  # Build map
+  leaflet() %>% 
+    #Add Basemap
+    addProviderTiles("Esri.WorldGrayCanvas", group = c("No labels", "Labels"))  %>%
+    addProviderTiles("CartoDB.PositronOnlyLabels", group = "Labels") %>%
+    
+    # Add data layers
+    addPolygons(data = p3_huc8_sf, group = "Subbasin (HUC 8)",
```

Comment:
Please include in this lake the name of the lake each huc8 is associated with, in addition to the existing label